plugins/api-docs/src/components/GraphQlDefinitionWidget/GraphQlDefinition.tsx

```diff
@@ -46,17 +53,46 @@ type Props = {
 
 export const GraphQlDefinition = ({ definition }: Props) => {
   const classes = useStyles();
+  const { entity } = useEntity();
   const schema = buildSchema(definition);
 
+  const url = entity.metadata.annotations?.['backstage.io/api-graphql-url'];
+
   return (
     <div className={classes.root}>
       <div className={classes.graphiQlWrapper}>
         <GraphiQL
-          fetcher={() => Promise.resolve(null) as any}
+          fetcher={
+            url
+              ? async (params: any, options: any = {}) => {
+                  const body = JSON.stringify(params);
+                  const headers = {
+                    'Content-Type': 'application/json',
+                    ...options.headers,
+                  };
+
+                  const res = await fetch(url!, {
+                    method: 'POST',
+                    headers,
+                    body,
+                  });
+                  return res.json();
+                }
+              : () => Promise.resolve(null) as any
+          }
           schema={schema}
           docExplorerOpen
           defaultSecondaryEditorOpen={false}
+          headerEditorEnabled
```

Comment:
is this an unrelated fix?

Reply:
this is to ensure the header editor always enabled, so we can pass some auth information through header.